checks/shell_download_validate.go

```diff
@@ -401,7 +400,63 @@ func isPipUnpinnedDownload(cmd []string) bool {
 		}
 	}
 
-	return isInstalled
+	return isInstall
+}
+
+func isPythonCommand(cmd []string) bool {
+	for _, pi := range pythonInterpreters {
+		if isBinaryName(pi, cmd[0]) {
+			return true
+		}
+	}
+	return false
+}
+
+func isPythonPipInstall(cmd []string) bool {
+	if !isPythonCommand(cmd) {
+		return false
+	}
+
+	isInstall := false
+	hasPip := false
+
+	for i := 1; i < len(cmd); i++ {
+		// Search for pip module.
+		if strings.EqualFold(cmd[i], "-m") &&
+			i < len(cmd)-1 &&
+			strings.EqualFold(cmd[i+1], "pip") {
+			hasPip = true
+			continue
+		}
+
+		// Search for install commands.
+		if strings.EqualFold(cmd[i], "install") {
+			isInstall = true
+			continue
+		}
+
+		if isInstall && hasPip {
```

Comment:
what's actually checking here that the install is pinned?

Reply:
you're absolutely right. I lost track of `python -m pip install -r _some_file`. Thank yoou!
Fixed and added it to unit tests.